lib/createStylelint.mjs

```diff
@@ -12,6 +12,7 @@ const STOP_DIR = IS_TEST ? process.cwd() : undefined;
  * @type {import('stylelint').PublicApi['_createLinter']}
  */
 export default function createStylelint(options = {}) {
+	// [INSERT HERE] CJS deprecation code
```

Comment:
major

Should we have one [here](https://github.com/stylelint/stylelint/blob/b0c7876d21a1900aaa3edfee6aa6e878bd4822e0/lib/augmentConfig.mjs#L324-L325)
```js
                // Handle either ES6 or CommonJS modules
                pluginImport = pluginImport.default || pluginImport;
```
as well?

Reply:
Your concern makes sense, and we should show the warning when importing a CJS plugin. But unfortunately, we cannot distinguish ESM and CJS by `pluginImport.default`: both modules should have the `default` property.

Is there any good solution?

Reply:
> Is there any good solution?

Since the path for CJS uses files with `.cjs` extensions, isn't it just a matter of adding the `if` only for `augmenConfig.cjs`?

Reply:
`.cjs` checks sounds good. Added it with 6434e98ce.

For `__esModule` check, it's non-standard, right? I'm concerned about supporting non-standard flags.

Reply:
ref: nodejs/node#40891